src/sympc/tensor/mpc_tensor.py

```diff
@@ -768,15 +817,8 @@ def __apply_op(
 
         result.shape = MPCTensor._get_shape(op_str, self.shape, y_shape, **kwargs_)
 
-        if op_str in {"mul", "matmul", "conv2d", "conv_transpose2d"} and (
-            not is_private or self.session.nr_parties > 2
-        ):
-            # For private op we do the division in the mul_parties function from spdz
-            scale = (
-                self.session.config.encoder_base
-                ** self.session.config.encoder_precision
-            )
-            result = result.truediv(scale)
+        # Check operation and apply truncation if required.
+        result = self.truncation(result, op_str, is_private)
```

Comment:
NITPICK: The method name should be a verb, not a noun. Could we rename this to ```self.truncate```?